packages/react/menu/src/Menu.tsx

```diff
@@ -361,6 +471,89 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
 
 MenuContent.displayName = CONTENT_NAME;
 
+/* -------------------------------------------------------------------------------------------------
+ * MenuSubContent
+ * -----------------------------------------------------------------------------------------------*/
+
+const SUB_CONTENT_NAME = 'MenuSubContent';
+
+type MenuSubContentOwnProps = Polymorphic.Merge<
+  Omit<
+    Polymorphic.OwnProps<typeof MenuContentImpl>,
+    'portalled' | 'disableOutsidePointerEvents' | 'disableOutsideScroll' | 'side' | 'trapFocus'
+  >,
+  {
+    /**
+     * Used to force mounting when more control is needed. Useful when
+     * controlling animation with React animation libraries.
+     */
+    forceMount?: true;
+  }
+>;
+type MenuSubContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  MenuSubContentOwnProps
+>;
+
+const MenuSubContent = React.forwardRef((props, forwardedRef) => {
+  const { forceMount, dir, ...contentProps } = props;
+  const context = useMenuContext(SUB_CONTENT_NAME);
+  const subMenuContext = useMenuSubContext(SUB_CONTENT_NAME);
+  const direction = dir || context.dir;
+  const timerRef = React.useRef(0);
+
+  React.useEffect(() => {
+    return () => window.clearTimeout(timerRef.current);
+  }, []);
+
+  return (
+    <Presence present={forceMount || context.open}>
+      <CollectionSlot>
+        <MenuContentImpl
+          id={subMenuContext.contentId}
+          align="start"
+          {...contentProps}
+          side={direction === 'ltr' ? 'right' : 'left'}
+          portalled
+          disableOutsidePointerEvents={false}
+          disableOutsideScroll={false}
+          trapFocus={false}
+          ref={composeRefs(forwardedRef, subMenuContext.contentRef)}
+          dir={direction}
+          onKeyDown={composeEventHandlers(contentProps.onKeyDown, (event) => {
+            const tabNext = event.key === 'Tab' && !event.shiftKey;
+            if (SUB_CLOSE_KEYS[direction].includes(event.key) || tabNext) {
+              // close a single level unless it's a forward tab, in which case close the entire stack
+              if (!tabNext) {
+                event.stopPropagation();
+              }
+              timerRef.current = window.setTimeout(() => {
+                subMenuContext.triggerElement?.focus();
+              }, 0);
```

Comment:
I was just trying to test this and couldn't seem to get it working: 

https://user-images.githubusercontent.com/175330/118336449-a0ada600-b509-11eb-978b-860ddfe17f89.mp4

I realised after it works if you never use the mouse though. Feels like a bug?

Reply:
Yeah it looks like the tab behaviour is more linked to `MenuBar`.

Reply:
I was a little unsure after looking at the spec plus some different implementations. In the context of `DropdownMenu` what would the expected behaviour be when we're inside a submenu? should tabbing close all subs and refocus the root menu? or close the entire menu stack (including root) and focus the button?

The focus trapping in `DropdownMenu` threw me because it doesn't let you tab out.

Reply:
I feel like generally these widgets are supposed to function as one tab stop. So once your widget is "focused" you use other keyboard navigation means (arrow keys, space, enter, escape, etc). So I feel like pressing tab or shift tab just shouldn't do anything.

It seems to be what MacOS does:

https://user-images.githubusercontent.com/1539897/118475196-d918d400-b703-11eb-9654-74d80c07cf94.mov

> Note: I'm pressing tab and shift-tab at the end.